### PR TITLE
{CI} Update pipeline to use Python 3.12

### PR DESCRIPTION
**Pull request by bebound**
Use 3.12 in CI as https://github.com/Azure/azure-cli/pull/29887 is merged

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: azure-pipelines.yml [hunk 1]
```diff
@@ -70,13 +70,13 @@ jobs:
...
-    displayName: 'Use Python 3.11'
+    displayName: 'Use Python 3.12'
     inputs:
-      versionSpec: 3.11
+      versionSpec: 3.12
   - bash: |
       #!/usr/bin/env bash
       set -ev
-      pip install wheel==0.30.0 requests packaging
+      pip install wheel==0.30.0 requests packaging setuptools
```
Comment (bebound) [inline azure-pipelines.yml:79]: wheel 0.30.0 is not compatible with Python 3.12. Install `setuptools` as a workaround.
```
Traceback (most recent call last):
  File "/mnt/vss/_work/1/s/./scripts/ci/test_index.py", line 23, in <module>
    from wheel.install import WHEEL_INFO_RE
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/wheel/install.py", line 17, in <module>
    from .paths import get_install_paths
  File "/opt/hostedtoolcache/Python/3.12.6/x64/lib/python3.12/site-packages/wheel/paths.py", line 7, in <module>
    import distutils.command.install as install
ModuleNotFoundError: No module named 'distutils'
```

**Review by wangzelin007 — APPROVED** [review]

File: azure-pipelines.yml (hunk 1 above)
Comment (jiasli) [reply azure-pipelines.yml:79]: Why installing `setuptools` can work around it?

Comment (bebound) [reply azure-pipelines.yml:79]: `setuptools` provides a "vendored" `distutils`.
```
root@7a5d336f2f35:/# python -c "import distutils;print(distutils.__path__)"
['/usr/local/lib/python3.12/site-packages/setuptools/_distutils']
```

**Review by jiasli — APPROVED** [review]

**Pull request merged**